### PR TITLE
Inclusion of per-key note selection to piano roll editor.

### LINKED ISSUE #529: Select/mark notes from piano-roll-key-context menu
A simple yet powerfull way to get all (especially bass) notes in a piano-roll container marked for further use.
The user rightclick a key in pianoroll, and get a new menu-item "Select all notes on this key"
This should be possible n times, and selected notes should remain selected. The user can now copy this extract of notes to a different instrument -for instance an effect bass. This would be much simpler, than doing several drag-ctrl+sh selections over many bars.

### PR DESCRIPTION
**Pull request by waxspin**
This is my addition of the functionality discussed in #529.

Closes #529

### CODE REVIEW

**Comment by waxspin** [comment]
The key changes for this added feature are in commit b931494c2c8c05b3250f9623ff565c23396358fb

**Comment by tresf** [comment]
@waxspin can you please rebase your commit history so that the other non-related commits don't show up in the commit history?  Ideally this would be a single commit.

Also, we'll need one person to confirm correct behavior prior to merging.

**Comment by waxspin** [comment]
@tresf Yes, I will sort out the commit history tonight-- sorry about that.  I will attempt to make a windows build as well for @musikBear, but don't have a windows environment set up yet, so that might take a few days if I don't find someone tonight to check it out.

**Comment by Umcaruje** [comment]
>I will attempt to make a windows build as well for @musikBear, but don't have a windows environment set up yet, so that might take a few days if I don't find someone tonight to check it out.

@waxspin @musikBear I might be able to make you a windows build tonight or tommorow - do you need a 32 or a 64 bit build?

**Comment by tresf** [comment]
> @tresf Yes, I will sort out the commit history tonight-- sorry about that. I will attempt to make a windows build as well for @musikBear, but don't have a windows environment set up yet, so that might take a few days if I don't find someone tonight to check it out.

None needed, just some cross-compiling packages for Ubuntu.  If you're not on Ubuntu, then @Umcaruje making a 32-bit one for @musikbear would be ideal.

**Comment by waxspin** [comment]
@tresf Thanks for the advice-- I have fixed my commit history here, and amended my notes accordingly.  @Umcaruje Thanks for the offer-- I will let you know if I need you to do that, but I'm going to try doing the windows build myself tonight using the link @tresf provided.

**Comment by tresf** [comment]
> I'm going to try doing the windows build myself tonight using the link @tresf provided.

If you need a place to host it, I'd recommend using https://github.com/waxspin/lmms/releases (and create a new release).  Just make sure to drag-and-drop it, as the upload button won't allow the 25MB attachment whereas the drag-and-drop will. :+1:

**Comment by waxspin** [comment]
Good to know, thanks!

**Comment by musikBear** [comment]
@waxspin -one is up already - like me to run tests on xp?

File: include/PianoRoll.h [hunk 1]
```diff
@@ -198,7 +199,8 @@ protected slots:
 		stmaUnmarkAll,
 		stmaMarkCurrentSemiTone,
 		stmaMarkCurrentScale,
-		stmaMarkCurrentChord,
+        stmaMarkCurrentChord,
```
Comment (tresf) [inline include/PianoRoll.h]: I think the tab formatting is wrong here as well as the following line.

Comment (waxspin) [inline include/PianoRoll.h]: Ah ok, that's annoying.  I think I saw that but forgot about it in the
process of setting up my windows build, so I can fix that today.

Comment (tresf) [inline include/PianoRoll.h]: Sounds like this is ready to go in once this is fixed.  Just bump the PR so we know it is ready.

**Comment by waxspin** [comment]
@tresf @musikBear This formatting has been fixed, I think you can pull it in now.

**Pull request merged**